src/web/app/components/sessions-table/sessions-table.component.html

```diff
@@ -61,60 +61,68 @@
+                        [disabled]="!sessionsTableRowModel.instructorPrivilege.canModifySession" (click)="unpublishSession(idx)">Unpublish Results</button>
+                <ng-template #publishButton>
+                  <button id="btn-publish-{{ idx }}" class="btn dropdown-item clickable" [disabled]="[FeedbackSessionSubmissionStatus.NOT_VISIBLE, FeedbackSessionSubmissionStatus.VISIBLE_NOT_OPEN].includes(sessionsTableRowModel.feedbackSession.submissionStatus)
+                    || sessionsTableRowModel.feedbackSession.publishStatus === FeedbackSessionPublishStatus.PUBLISHED || !sessionsTableRowModel.instructorPrivilege.canModifySession"
+                          ngbTooltip="Make session responses available for viewing" placement="left" container="body" (click)="publishSession(idx)">Publish Results</button>
+                </ng-template>
+                <button id="btn-resend-{{ idx }}" class="btn dropdown-item clickable" *ngIf="sessionsTableRowModel.feedbackSession.publishStatus === FeedbackSessionPublishStatus.PUBLISHED" (click)="remindResultsLinkToStudent(idx)">Resend link to view results</button>
+                <button id="btn-download-{{ idx }}" class="btn dropdown-item clickable" (click)="downloadSessionResults(idx)"
+                        [disabled]="!sessionsTableRowModel.instructorPrivilege.canViewSessionInSections">Download Results</button>
+              </div>
             </div>
-          </div>
-          <div ngbDropdown class="d-inline-block" ngbTooltip="Send e-mails to remind students and instructors who have not submitted their feedbacks to do so">
-            <button id="btn-remind-{{ idx }}" class="btn btn-light btn-sm align-row" (click)="sendRemindersToStudents(idx); setRowClicked(idx)"
-                    [disabled]="sessionsTableRowModel.feedbackSession.submissionStatus !== FeedbackSessionSubmissionStatus.OPEN ||
-                          !sessionsTableRowModel.instructorPrivilege.canModifySession">
-              <span>Remind</span>
-              <div class="padding-left-7px">
-                <tm-ajax-loading [color]="'#000000'" *ngIf="isSendReminderLoading && rowClicked === idx"></tm-ajax-loading>
+            <div ngbDropdown class="d-inline-block" ngbTooltip="Send e-mails to remind students and instructors who have not submitted their feedbacks to do so">
+              <button ngbDropdownToggle id="btn-remind-{{ idx }}" class="btn btn-light btn-sm"
+                      [disabled]="sessionsTableRowModel.feedbackSession.submissionStatus !== FeedbackSessionSubmissionStatus.OPEN ||
+                            !sessionsTableRowModel.instructorPrivilege.canModifySession">
+                <div class="d-inline-flex">
+                  <span>Remind</span>
+                  <div class="padding-left-7px">
+                    <tm-ajax-loading [color]="'#000000'" *ngIf="rowClicked === idx"></tm-ajax-loading>
```

Comment:
Any reason for removing "isSendReminderLoading" from the `*ngIf` condition? Without this the loading spinner does not go away after the modal is open.

Reply:
It slipped my mind to add it back during testing, good catch!